kivy/uix/behaviors/compoundselection.py

```diff
@@ -308,7 +308,7 @@ def select_with_touch(self, node, touch=None):
             return False
 
         if (node in self.selected_nodes and (not range_select)):  # selected
-            if multiselect:
+            if ((not multi) or multiselect):
                 self.deselect_node(node)
             else:
                 self.clear_selection()
```

Comment:
The goal of a fix would be to be able to deselect the last selected item regardless of the value of `touch_multiselect` and `multiselect`, while still handling edge cases, one of them being the one you have found.